zulipterminal/ui_tools/boxes.py

```diff
@@ -63,9 +64,11 @@ def private_box_view(self, button: Any=None, email: str='') -> None:
         ]
         self.focus_position = 1
 
-    def stream_box_view(self, caption: str='', title: str='') -> None:
+    def stream_box_view(self, stream_id: int, caption: str='', title: str=''
+                        ) -> None:
         self.set_editor_mode()
         self.to_write_box = None
+        self.stream_id = stream_id
```

Comment:
This gets set only on set-up, so if the stream is changed before the message is sent, the topic autocomplete uses the stream used at initialization. This is the same issue present with another PR, maybe?

Reply:
Yes, I can fix it now that #738 is merged

Reply:
This is resolved by the change at the end  of this commit?

Reply:
Yes that should keep `stream_id` in sync